internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -146,59 +153,65 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
-// If the driver name is not registered in the connection pool, it returns
-// false and requeues the request.
-func (r *PersistentVolumeClaimReconciler) checkDriverSupportReclaimsSpace(logger *logr.Logger, annotations map[string]string, driver string) (bool, bool) {
-	reclaimSpaceSupportedByDriver := false
-
-	if drivers, ok := annotations[rsCSIAddonsDriverAnnotation]; ok && slices.Contains(strings.Split(drivers, ","), driver) {
-		reclaimSpaceSupportedByDriver = true
-	}
-
-	ok := r.supportsReclaimSpace(driver)
-	if reclaimSpaceSupportedByDriver && !ok {
-		logger.Info("Driver supports spacereclamation but driver is not registered in the connection pool, Reqeueing request", "DriverName", driver)
-		return true, false
-	}
-
-	if !ok {
-		logger.Info("Driver does not support spacereclamation, skip Requeue", "DriverName", driver)
+// checkDriverSupportCapability checks if the given driver supports the specified capability.
+// It is also used to determine the requeue in case the driver supports the capability
+// but the capability is not found in connection pool.
+// The first return value is true when a requeue is needed.
+// The second return value is true when the capability is registered.
+func (r *PersistentVolumeClaimReconciler) checkDriverSupportCapability(
+	logger *logr.Logger,
+	annotations map[string]string,
+	driverName string,
+	cap Operation) (bool, bool) {
+	driverSupportsCap := false
+	capFound := false
+
+	var driverAnnotation string
+	switch cap {
+	case relciamSpaceOp:
+		driverAnnotation = rsCSIAddonsDriverAnnotation
+	case keyRotationOp:
+		driverAnnotation = krCSIAddonsDriverAnnotation
+	default:
+		logger.Info("Unknown capability", "Capability", cap)
 		return false, false
 	}
 
-	return false, true
-}
+	if drivers, ok := annotations[driverAnnotation]; ok && slices.Contains(strings.Split(drivers, ","), driverAnnotation) {
+		driverSupportsCap = true
+	}
 
-// checkDriverSupportsEncryptionKeyRotate checks if the driver supports key
-// rotation or not. If the driver does not support key rotation, it
-// returns false and if the driver supports key rotation, it returns true.
-// If the driver name is not registered in the connection pool, it returns
-// false and requeues the request.
-func (r *PersistentVolumeClaimReconciler) checkDriverSupportsEncryptionKeyRotate(
-	logger *logr.Logger,
-	annotations map[string]string,
-	driver string) (bool, bool) {
-	keyRotationSupportedByDriver := false
+	conns := r.ConnPool.GetByNodeID(driverName, "")
+	for _, conn := range conns {
+		for _, c := range conn.Capabilities {
+			switch cap {
+			case relciamSpaceOp:
+				capFound = c.GetReclaimSpace() != nil
+			case keyRotationOp:
+				capFound = c.GetEncryptionKeyRotation() != nil
+			default:
+				continue
+			}
 
-	if drivers, ok := annotations[krCSIAddonsDriverAnnotation]; ok && slices.Contains(strings.Split(drivers, ","), driver) {
-		keyRotationSupportedByDriver = true
+			if capFound {
+				return false, true
+			}
```

Comment:
you don't need to check capFound in below lines.

Reply:
Done. Here's a diff of the latest change for ease of review:
```diff
-       if driverSupportsCap && !capFound {
-               logger.Info(fmt.Sprintf("Driver supports %s but driver is not registered in the connection pool, Reqeueing request", cap), "DriverName", driverName)
+       // If the driver supports the capability but the capability is not found in connection pool,
+       if driverSupportsCap {
+               logger.Info(fmt.Sprintf("Driver supports %s but driver is not registered in the connection pool, Requeuing request", cap), "DriverName", driverName)
                return true, false
        }

-       if !capFound {
-               logger.Info(fmt.Sprintf("Driver does not support %s, skip Requeue", cap), "DriverName", driverName)
-               return false, false
-       }
-
-       // Do not reque, cap is supported
-       return false, true
-
+       // If the driver does not support the capability, skip requeue
+       logger.Info(fmt.Sprintf("Driver does not support %s, skip Requeue", cap), "DriverName", driverName)
+       return false, false
 }
```